### PR TITLE
Enhance Holdover Management and Fault Reporting and holdover in ts2phc/Dpll

### PR DESCRIPTION
**Pull request by aneeshkp**
1. Removed NMEA lost updates from metrics and added fault reporting. The ts2phc holdover feature will now manage ts2phc to continue running even when NMEA updates are lost.
2. Added logic to handle Holdover with consideration for whether frequency traceability is available (SyncE enabled) or not. 

Three variables are used in the PTP configuration to manage Holdover: `LocalMaxHoldoverOffset`, `LocalHoldoverTimeout`, and `MaxInSpecOffset`. 

- Holdover timing transitions based on whether frequency traceability is enabled:
  - If SyncE is not enabled (frequency traceability is false), a slope calculation is used to determine the time to reach `MaxInSpecOffset`.
  - If SyncE is enabled, `LocalHoldoverTimeout` is used to calculate the total holdover duration.
  
Additionally, updated the logic for clock class transitions:
- Changed the behavior to announce a transition to FREE RUN from class 7 when `MaxInSpecOffset` is reached and SyncE is not available.
- Added logic to manage ClockClass 140 when crossing `MaxInSpecOffset` with SyncE enabled.

### CODE REVIEW

**Review by vitus133 — COMMENTED** [review]
Hi,
Added a couple of comments for "nmea string lost" logic and slope units.
Otherwise looks good.
Thanks!

File: pkg/daemon/daemon.go [hunk 1]
```diff
@@ -965,15 +970,6 @@ func (p *ptpProcess) processPTPMetrics(output string) {
...
 		p.ProcessSynceEvents(logEntry)
-
-	} else if p.name == ts2phcProcessName &&
-		(strings.Contains(output, NMEASourceDisabledIndicator) ||
-			strings.Contains(output, InvalidMasterTimestampIndicator) ||
-			strings.Contains(output, NMEASourceDisabledIndicator2)) &&
-		!leap.LeapMgr.IsLeapInWindow(time.Now().UTC(), -2*time.Second, time.Second) { //TODO identify which interface lost nmea or 1pps
-		iface := p.ifaces.GetGMInterface().Name
-		p.ProcessTs2PhcEvents(faultyOffset, ts2phcProcessName, iface, state, map[event.ValueType]interface{}{event.NMEA_STATUS: int64(0)})
-		glog.Error("nmea string lost") //TODO: add for 1pps lost
```
[the remark this answers is not in the document]
Comment (aneeshkp) [reply pkg/daemon/daemon.go:976]: no it has no value for now , we will revisit later

File: pkg/dpll/dpll.go [hunk 2]
```diff
@@ -939,6 +946,6 @@ func CalculateTimer(nodeProfile *ptpv1.PtpProfile) (int64, int64) {
 		}
 	}
 	slope := float64(localMaxHoldoverOffSet) / float64(localHoldoverTimeout) * 1000.0
-	timer := int64(math.Round(float64(maxInSpecOffset) * 1000 / slope))
-	return int64(localMaxHoldoverOffSet), timer
+	inSpecTimer := int64(math.Round(float64(maxInSpecOffset) * 1000 / slope))
+	return int64(maxInSpecOffset), int64(localMaxHoldoverOffSet), int64(localHoldoverTimeout), inSpecTimer, false
```
[the remark this answers is not in the document]
Comment (aneeshkp) [reply pkg/dpll/dpll.go:950]: right  , I should keep it in ns/s , you need to convert it to ps/s

File: pkg/daemon/daemon.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (vitus133) [reply pkg/daemon/daemon.go:976]: Ah, is it because of ts2phc holdover?

**Pull request merged**